samples/bluetooth/hci_spi/src/main.c

```diff
@@ -195,22 +199,24 @@ static void bt_tx_thread(void *p1, void *p2, void *p3)
 
 		switch (rxmsg[PACKET_TYPE]) {
 		case HCI_CMD:
-			buf = bt_buf_get_tx(BT_BUF_CMD, K_NO_WAIT, &rxmsg[1],
-					    sizeof(cmd_hdr));
+			buf = bt_buf_get_tx(BT_BUF_CMD, K_NO_WAIT,
+					    hci_hdr.cmd_hdr,
+					    sizeof(*hci_hdr.cmd_hdr));
 			if (buf) {
 				net_buf_add_mem(buf, &rxmsg[4],
-						cmd_hdr.param_len);
+						hci_hdr.cmd_hdr->param_len);
 			} else {
 				LOG_ERR("No available command buffers!");
 				continue;
 			}
 			break;
 		case HCI_ACL:
 			buf = bt_buf_get_tx(BT_BUF_ACL_OUT, K_NO_WAIT,
-					    &rxmsg[1], sizeof(acl_hdr));
+					    hci_hdr.acl_hdr,
+					    sizeof(*hci_hdr.acl_hdr));
```

Comment:
I further reduced the number of reference to &rxmsg[1] by using the union pointer.